### ISSUE TITLE
Does rasa core keras policy training change? It seems less good than rasa_core before

### ISSUE THREAD

**Issue by fakesecret** [RasaHQ/rasa#3637]
**Rasa version**: Rasa 1.0
**Python version**: 3.7
**Operating system** (windows, osx, ...): Linux
**Issue**: (question) Does rasa core keras policy training change? It seems less good than rasa_core before.

While using the same training data (i.e. stories) and policy config as rasa_core before (Keras + Memoization), the action taken by my chatbot changed dramatically. Saying that I only have 3 stories like "story1: intent_A -> utter_A; story2: intent_B -> utter_B; story3: intent_C -> utter_C".

When using rasa_core before, I obtained dialogue like "intent_A -> utter_A -> intent_B -> utter_B -> intent_A -> utter A". 
However, now using rasa 1.0, I obtain "intent_A -> utter_A -> intent_B -> ??" and "intent_A -> utter_A -> intent_A -> ??".  (?? means no action or action with extremly low confidence) It seems rasa now consider with higher weight the beginning and the end of stories.

Edit: 
**Reason 1**: Augmentation (def=20, in rasa 1.0.3) is a bit reduced for KerasPolicy. By increasing augmentation factor, Chatbot can answer correctly in the situations like "intent_A -> utter_A -> intent_A -> utter_A", etc. Because by augmentation, the initial stories are glued together as new training stories.
**Reason 2**: As I first tested with "rasa init" using all default options (random seed is not set), the model is underfit. Different runs result in different performances. By increasing epochs (ex: 500) and lowering batch_size (ex: 20), I obtained a much more converged model.

**Comment by akelad** [comment]
Nothing has changed between versions about how the KerasPolicy was implemented. The only thing that has changed is that the augmentation doesn't get applied to the MemoizationPolicy and also it's a bit reduced for KerasPolicy. So by default it will train on `50` stories that are created from your training data. You could probably increase the augmentation factor, but i'm not sure that's really necessary with so few stories.

Could you post the output of `rasa train --debug` for the core part?

**Comment by fakesecret** [comment]
@akelad Thanks for the info. I did many tries just now. Now I know why. This is not because of Keras Policy change.

It is because the "rasa init" example has KerasPolicy by default. These default options are not enough for the core model to converge. So runs can result in very differernt models, randomly.

With the same config (all by default as "rasa init"), I have Run 1:
```
Epoch 100/100
62/62 [==============================] - 0s 148us/sample - loss: 1.0351 - acc: 0.5806
```
Run 2:
```
Epoch 100/100
62/62 [==============================] - 0s 243us/sample - loss: 0.8892 - acc: 0.8548
```

When I set epochs: 500 and batch_size:20, we have a converged model and the performance is much better. So, for me, if you can make the "rasa init" example more stable, it should be better.

**Comment by fakesecret** [comment]
@akelad Hello again. You were right. I did more tests.

Two reasons make this happen. First reason is the default options (epochs, batches, etc.) are trying to underfit like I explained above in the example of "rasa init". Because I didn't set random seed, the results varied a lot. The other reason is that the augmentaion is reduced. That means "intent_A -> utter_A -> intent_B -> utter_B -> intent_A -> utter A" has lower chance of being created in the data augmentation step. Chatbot tends to answer correctly to the first query, but incorrect if the following query is not in a training story.

**Comment by akelad** [comment]
Awesome glad you managed to make this work

**Comment by sushilr007** [comment]
@Jinchao-Yu Hello I am curious about how did you do that? Can you please share your config.yml file?
Thank you.

**Comment by Jinchao-Yu** [comment]
@sushilr007 Do you mean this file?
```
policies:
  - name: "KerasPolicy"
    featurizer:
    - name: MaxHistoryTrackerFeaturizer
      max_history: 5
      state_featurizer:
        - name: BinarySingleStateFeaturizer
    epochs: 400
    batch_size: 50
    validation_split: 0.2
  - name: "MemoizationPolicy"
    max_history: 5
  - name: "FallbackPolicy"
    nlu_threshold: 0.2
    core_threshold: 0.2
    fallback_action_name: "utter_default"
```
Augmentation factor can be configured in a script for training rasa core. Maybe it can be set in command line too but I am not sure because I always use my scripts for training. 
```python3
loop = asyncio.get_event_loop()
    training_data = loop.run_until_complete(
        agent.load_data(dir_core_training_data,
                        augmentation_factor=CORE_AUGMENTATION_FACTOR)
    )  # -> List[DialogueStateTracker]
    agent.train(training_data)
```

**Comment by sushilr007** [comment]
Thank you @Jinchao-Yu I exactly want this.  I read your issue and same I was thinking on that.
and also please share, which pipeline you are using?
Thank you so much.

**Comment by Jinchao-Yu** [comment]
@sushilr007 it's better for you to test your training sets with several pipelines based on your needs and select the best. For ex, I have my own class of preprocessing, which can change the choice of tokenization. Mine is:
```
language: "en"

pipeline:
- name: "SpacyNLP"
- name: "SpacyTokenizer"
- name: "CRFEntityExtractor"
  # "pos", "pos2" can be applied only when using SpacyNLP
  features: [
  ["low", "title", "upper"],
  ["bias", "low", "prefix5", "prefix2", "suffix5", "suffix3", "suffix2", "upper", "title", "digit", "pattern", "pos", "pos2"],
  ["low", "title", "upper"]]
- name: "CountVectorsFeaturizer"
- name: "EmbeddingIntentClassifier"
  intent_tokenization_flag: true
  intent_split_symbol: "+"
```
